ocp_resources/resource.py

```diff
@@ -112,15 +116,34 @@ def get_client(
         # If `KUBECONFIG` environment variable is set via code, the `KUBE_CONFIG_DEFAULT_LOCATION` will be None since
         # is populated during import which comes before setting the variable in code.
         config_file = config_file or os.environ.get("KUBECONFIG", "~/.kube/config")
+
+        if os.environ.get("OPENSHIFT_PYTHON_WRAPPER_CLIENT_USE_PROXY"):
+            proxy = os.environ.get("HTTPS_PROXY") or os.environ.get("HTTP_PROXY")
+            if not proxy:
+                raise ValueError(
+                    "Proxy configuration is enabled but neither HTTPS_PROXY nor HTTP_PROXY environment variables are set."
+                )
+            if client_configuration.proxy and client_configuration.proxy != proxy:
+                raise ValueError(
+                    f"Conflicting proxy settings: client_configuration.proxy={client_configuration.proxy}, "
+                    f"but the environment variable 'OPENSHIFT_PYTHON_WRAPPER_CLIENT_USE_PROXY' defines proxy as {proxy}."
```

Comment:
This message is misleading the value of proxy is not coming from OPENSHIFT_PYTHON_WRAPPER_CLIENT_USE_PROXY

Reply:
I'm going to merge the PR and I will fix the error in follow up PR